### ISSUE TITLE
Add an option to announce extra routes

### ISSUE THREAD

**Issue by Gyosa3** [lmagyar/homeassistant-addon-tailscale#40]
Hi,

I'm wondering if you would like to add an extra configuration option where one can define the routes to announce. I am not fancy of the "auto-discovery" function built in the addon and I'd like to override it when needed.

Something in the area of:

- when no option is set, autodiscover the local LAN where HA is running
- when the option is set to "disabled" (whatever way), do not announce any route
- when option is set, do not autodiscover and instead announce the routes that are set in the option

If you prefer I can open an issue in the official addon repo.

**Comment by lmagyar** [comment]
FYI: I've made `snat_subnet_routes` option available, leaving this issue open as a reminder for `advertise_routes` option (I still see 0% chance, that an `advertise_routes` option can be merged in the official repo).

**Comment by Gyosa3** [comment]
Hi again,

ok I have tried a few things on the site where I have multiple VLANs and I can't figure out a way to make things work without anouncing the routes on the HA addon. Here is my setup ("VLAN 1" means 192.168.1.0/24 and "TS A" means "Tailscale instance A" within the VLAN)

```
Site A              Site B
--------------     --------------
VLAN 2 (TS A)       VLAN 5 (TS B)
VLAN 3
VLAN 4
------------
VLAN 25
------------
VLAN 95 (TS C)
```
TS A and TS B are your TS addons on HA instances with the restrictions explained above. TS C is a plain TS engine running on a dedicated Raspberry Pi Zero where I can run any command line of my choice.

My requirements:
R1 -  Any machine in Site B should access any machine on VLAN 2, 3 and 4 via TS B but not machines in VLAN 25 and 95.
R2 - Any machine on site A can access any machine on site B (VLAN 5) via TS A or TS C.
R3 - machines on VLAN 25 should access VLAN 2, 3 and 4 locally and exit to internet on the local router on site A
R4 - machines on VLAN 95 should access VLAN 2, 3 and 4 locally but exit to internet via TS C which is using TS B as exit point to internet.

If I do it by feeling, I would treat VLAN 2, 3 and 4 together as one sub-site and VLAN 95 as another sub-site, and use TS A and TS C to bridge the two together. **I can't do this because TS A is only announcing VLAN 2 (if I use yours or the official addon).**

I have tried to announce VLAN 3 and 4 on TS C but because it also uses an exit node it fails to route correctly the local routes.
I have tried to announce VLAN 2, 3 and 4 on TS C and disable TS A but it also fails for the same reason.

The furthest I've been is the following:
- I've disabled completely TS A and revoked the routes.
- I've announced VLAN 2, 3, 4 and 95 on TS C while using TS B as exit node by launching
`sudo tailscale up --advertise-routes=192.168.2.0/24,192.168.3.0/24,192.168.4.0/24,192.168.95.0/24  --snat-subnet-routes=false --accept-routes --exit-node=x.x.x.x --exit-node-allow-lan-access=true`
- I have managed to preserve access from local machines in VLAN 95 to other local subnets by executing
`sudo ip route add throw 192.168.2.0/24 table 52` (also for .3.0 and .4.0)

What I am missing is to route packets coming from tailscale to VLAN 2, 3 and 4. there seem to be an issue with the forwarding of packets from tailscale0 to eth0 for these VLANs and it feels very close to what you corrected recently for the site-to-site routing I think.

So I'd like to have your opinion on this, do you think I'm just missing a route on TS C and all is good or it's an impossible routing and I should reinstall TS A with an addon that can announce multiple routes?

thanks for your help :)

**Comment by lmagyar** [comment]
I'm on vacation, I'm on phone, I will return one week from now.

**Comment by lmagyar** [comment]
New v0.11.1.11 is released, it makes advertised subnet routes configurable.

Notes:
- Now you can advertise routes on TS A
  ```
  advertise_routes:
    - 192.168.2.0/24
    - 192.168.3.0/24
    - 192.168.4.0/24
  ```
- turn off `snat-subnet-routes` only when you really need this, it requires that you configure the routing settings on all local non-TS devices, because the incoming traffic will not originate from the TS node's local IP, but from a subnet on the other side
- until the subnets are working, don't use exit-node, don't try to configure/test 2 things in parallel
- table 52 is managed by Tailscale's, I don't think it is a good idea to change it manually

**Comment by Gyosa3** [comment]
Thank you so much !!!!!!!!! Hope you had great vacation :)

I'm testing it with the following configuration:
(HA is running AdGuard as DNS service for all local networks so I have to disable Tailscale DNS)

```
accept_dns: false
advertise_exit_node: false
userspace_networking: false
log_level: debug
advertise_routes:
  - 192.168.2.0/28
  - 192.168.3.0/24
  - 192.168.4.0/24
  - 192.168.25.0/24
snat_subnet_routes: false

```
1/ I can confirm that all 4 routes show up in the Tailscale admin page
![image](https://github.com/lmagyar/homeassistant-addon-tailscale/assets/51777889/165e2321-29be-4ee7-81e7-cf1cceb1e4ac)
2/ From any machine in any of the announced networks I can connect to the "5" network remotely via a static route to the local HA (I have to test what happens if I try with a machine in a local network not announced)
![image](https://github.com/lmagyar/homeassistant-addon-tailscale/assets/51777889/3f489654-eb35-4904-a350-a0effa336697)
Pinging from 192.168.25.xx on site A to HA instance on site B:
![image](https://github.com/lmagyar/homeassistant-addon-tailscale/assets/51777889/22a35e4b-96ee-4c92-8c98-2aef42315ec4)

3/ From a remote tailscale machine I can reach the local HA but **I cannot reach any device in the other announced routes**. From the route table I see an entry for the original local route (the "2" network). I would assume the other routes would fall into the "default" route but it seems not. I can't figure out why, and if it would be necessary to have one specified entry for each announced route?

![image](https://github.com/lmagyar/homeassistant-addon-tailscale/assets/51777889/550b1ae4-831a-4a68-9471-24a3f4e31c1d)

Nearly there! any idea? Thanks again!

**Comment by lmagyar** [comment]
Thank you for the reply, as I see it seems to be working finally, you tested thoroughly this site-to-site networking setup. :)

**Comment by Gyosa3** [comment]
Hi, two last comments about the setup for the record:

- I have tested with a non-tailscale machine located in the "5" remote network and I can reach any device from the other "2", "3", etc. networks without any problems, so it's confirmed it works both ways.
- **Cherry on the cake:** the two subrouter nodes located on the same site (so your HA addon and the tailscale setup on a Pi zero) are using Tailscale's capability to detect local routes. So my specific devices located in my segregated "95" network are using the subnet router from the remote "5" network as exit point, BUT they still route locally to the "2", "3", etc. local networks because Tailscale "knows" that it is local. I did not know that actually.
![image](https://github.com/lmagyar/homeassistant-addon-tailscale/assets/51777889/2ff4eace-b315-451b-8636-6c40f907ff47)
I therefore have full access to machines local and remote, with all expected routing constraints, while maintaining best performance overall. I could not wish better.

Thanks :)
